src/components/subcomponents/join-team-button.js

```diff
@@ -3,13 +3,13 @@ import { Button } from "@material-ui/core";
 import { joiningHackIdea } from "../../store/actions/userActions";
 import { connect } from "react-redux";
 
-export function JoinButton({ hackDetails, user, dispatch}) {
+export function JoinButton({ hackDetails, user, dispatch, history}) {
 
   const [disable, setDisable] = React.useState(false)
 
   const joinHack = event => {
     event.preventDefault();
-    dispatch(joiningHackIdea(user.userId, hackDetails._id));
+    dispatch(joiningHackIdea(user.userId, hackDetails._id, history, user.jwt));
```

Comment:
I think you should use the `getState` feature of the thunk to access the token from state, rather than passing it through (actions know they needs auth, not the components).

Reply:
I'll look into connected redux router and look into implementing it down the track.
I did not know that the state could be called inside the action